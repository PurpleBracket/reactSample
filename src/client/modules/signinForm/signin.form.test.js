/* global describe test expect */
import React from 'react'
import { shallow } from 'enzyme'
import SigninForm from './signin.form'

// beforeAll(() => {
//   global.logger = {
//     info: function () {
//
//     }
//     // info: jest.fn(),
//     // log: jest.fn(),
//     // error: jest.fn()
//   }
// })

describe('Register', () => {
  test('should match snapshot', () => {
    const wrapper = shallow(<SigninForm />)
    expect(wrapper).toMatchSnapshot()
  })

  test('email is stored in state', () => {
    const wrapper = shallow(<SigninForm />)
    let input = wrapper.find('[name="email"]')
    input.simulate('change', {target: {name: 'email', value: 'abc'}})
    expect(wrapper.state().email).toEqual('abc')
  })

    // test('handle submit calls login', async done => {
    //   let mockHistory = jest.fn()
    //   let mockLogin = jest.fn((state) => {
    //     return new Promise((resolve, reject) => {
    //       if (state) {
    //         resolve(state)
    //       } else {
    //         reject(new Error('Empty state object'))
    //       }
    //     })
    //   })
    //
    //   let wrapper = shallow(<Signin history={{push: mockHistory}} />)
    //   wrapper.instance().client.login = mockLogin
    //
    //   wrapper.update()
    //   wrapper.instance().handleSubmit({
    //     preventDefault: () => {}
    //   })
    //
    //   await expect(mockLogin).toBeCalled()
    //   await expect(mockHistory).toBeCalled()
    //   done()
    // })
})
