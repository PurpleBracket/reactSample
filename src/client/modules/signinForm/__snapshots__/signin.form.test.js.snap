// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register should match snapshot 1`] = `
<div>
  <div
    className="columns"
  >
    <div
      className="column"
    />
    <div
      className="column is-one-third"
      style={
        Object {
          "minHeight": "100vh",
        }
      }
    >
      <Panel
        className="signinForm"
        noBorder={true}
        noShadow={true}
        transparentBG={true}
      >
        <Subhead
          size="4"
        >
          Sign In
        </Subhead>
        <form
          className="card-content"
          onSubmit={[Function]}
        >
          <Input
            hasIconLeft={true}
            hasIconRight={true}
            icon="envelope"
            name="email"
            onChange={[Function]}
            placeholder="Email Address"
            required={true}
            type="email"
          />
          <Input
            hasIconLeft={true}
            hasIconRight={true}
            icon="unlock-alt"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
          />
          <div
            className="field buttons is-centered"
          >
            <Button
              color="white"
              style=""
              type="submit"
            >
              Sign In
            </Button>
          </div>
        </form>
        <footer
          className="card-footer"
        >
          <p
            className="card-footer-item"
          >
            <span>
              <LinkAtom
                href="/signup"
              >
                Don't have an account? 
                <br />
                 
                <b>
                  Create an account
                </b>
              </LinkAtom>
            </span>
          </p>
          <p
            className="card-footer-item"
          >
            <span>
              <LinkAtom
                href="/password-reset"
              >
                Forgot Password?
              </LinkAtom>
            </span>
          </p>
        </footer>
      </Panel>
    </div>
    <div
      className="column"
    />
  </div>
</div>
`;
